models/repo/repo.go

```diff
@@ -528,6 +528,18 @@ func (repo *Repository) ComposeCompareURL(oldCommitID, newCommitID string) strin
 	return fmt.Sprintf("%s/%s/compare/%s...%s", url.PathEscape(repo.OwnerName), url.PathEscape(repo.Name), util.PathEscapeSegments(oldCommitID), util.PathEscapeSegments(newCommitID))
 }
 
+func (repo *Repository) ComposeBranchCompareURL(baseRepo *Repository, branchName string) string {
+	var cmpBranch string
+	if baseRepo == nil {
+		baseRepo = repo
+	}
+	if repo.ID != baseRepo.ID {
+		cmpBranch = fmt.Sprintf("%s/%s:", url.PathEscape(repo.OwnerName), url.PathEscape(repo.Name))
+	}
+	cmpBranch = fmt.Sprintf("%s%s", cmpBranch, util.PathEscapeSegments(branchName))
+	return fmt.Sprintf("%s/compare/%s...%s", baseRepo.Link(), baseRepo.DefaultBranch, cmpBranch)
```

Comment:
Wait… `baseRepo.DefaultBranch`?
I don't think you always want to compare against that.
If we add another function parameter, this function will be a lot more useful for all other edge cases wanting it.

Reply:
When click the `create pull request` button, you can still change the target repository and branch. And any better branch can be used here?